File: ollama/codigo-final/binary-trees/codegemma-7b/fewshot/temp_0.7/binary-trees.py
import sys

class TreeNode:
    def __init__(self):
        self.left = TreeNode()
        self.right = TreeNode()

    def __del__(self):
        del self.left
        del self.right

def stretch_tree(depth):
    root = TreeNode()
    for _ in range(2 ** (depth - 1)):
        node = root
        for _ in range(depth - 1):
            node = node.left
        del node
    return True

def count_nodes(root):
    if root is None:
        return 0
    return 1 + count_nodes(root.left) + count_nodes(root.right)

def bottom_up_trees(depth, iterations):
    trees = [TreeNode() for _ in range(iterations)]
    for tree in trees:
        count_nodes(tree)
    del trees
    return True

def long_lived_tree(depth):
    root = TreeNode()
    for _ in range(2 ** (depth - 1)):
        node = root
        for _ in range(depth - 1):
            node = node.left
        del node
    return True

if __name__ == "__main__":
    N = int(sys.argv[1])
    min_depth = 4
    max_depth = max(min_depth + 2, N)
    stretch_depth = max_depth + 1

    for D in range(min_depth, max_depth + 1):
        iterations = (1 << (max_depth - D + min_depth))
        print(f"stretch tree of depth {D}\t check: {stretch_tree(stretch_depth)}")
        print(f"I\t trees of depth {D}\t check: {bottom_up_trees(D, iterations)}")

    print(f"long lived tree of depth {max_depth}\t check: {long_lived_tree(max_depth)}")

File: ollama/codigo-final/binary-trees/codegemma-7b/fewshot/temp_0.7/binary-trees.java
import java.util.Scanner;

public class BinaryTreeOperations {

    static class TreeNode {
        TreeNode left, right;
        TreeNode() {}
    }

    static int min_depth = 4;
    static int max_depth;
    static int stretch_depth;

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read N from command line
        int N = Integer.parseInt(args[0]);

        // Calculate max_depth
        max_depth = Math.max(min_depth + 2, N);
        stretch_depth = max_depth + 1;

        // Stretch Tree
        TreeNode stretchTree = createTree(stretch_depth);
        verifyNode(stretchTree);
        deleteTree(stretchTree);

        // Iterative Trees
        for (int depth = min_depth; depth <= max_depth; depth += 2) {
            int iterations = 1 << (max_depth - depth + min_depth);
            for (int i = 0; i < iterations; i++) {
                TreeNode iterativeTree = createTree(depth);
                countNodes(iterativeTree);
                deleteTree(iterativeTree);
            }
        }

        // Long-Lived Tree
        TreeNode longLivedTree = createTree(stretch_depth);
        verifyNode(longLivedTree);
        deleteTree(longLivedTree);
    }

    // Create a binary tree of the given depth
    private static TreeNode createTree(int depth) {
        if (depth == 0) {
            return null;
        }
        TreeNode node = new TreeNode();
        node.left = createTree(depth - 1);
        node.right = createTree(depth - 1);
        return node;
    }

    // Verify that the tree node exists
    private static void verifyNode(TreeNode node) {
        if (node == null) {
            System.out.println("check: false");
        } else {
            System.out.println("check: true");
        }
    }

    // Delete the binary tree
    private static void deleteTree(TreeNode node) {
        if (node == null) {
            return;
        }
        deleteTree(node.left);
        deleteTree(node.right);
        node = null;
    }

    // Count the nodes in the binary tree
    private static int countNodes(TreeNode node) {
        if (node == null) {
            return 0;
        }
        return 1 + countNodes(node.left) + countNodes(node.right);
    }
}
